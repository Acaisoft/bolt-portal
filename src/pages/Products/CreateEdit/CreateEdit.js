import React, { Component } from 'react'
import PropTypes from 'prop-types'

export class CreateEdit extends Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string,
      }).isRequired,
    }).isRequired,
  }

  render() {
    const { match } = this.props
    const { id } = match.params

    return (
      <div>{id ? `Editing product with ID=${id}` : 'Creating a new product'}</div>
    )
  }
}

export default CreateEdit
