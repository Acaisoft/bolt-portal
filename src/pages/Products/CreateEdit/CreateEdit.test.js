import React from 'react'
import { shallow } from 'enzyme'

import { CreateEdit } from './CreateEdit'

const initCreateEdit = overrides => {
  const mockProps = {
    match: {
      params: {},
    },
  }
  const wrapper = shallow(<CreateEdit {...mockProps} {...overrides} />)
  return { wrapper }
}

describe('page: Products/CreateEdit', () => {
  describe('rendering', () => {
    it('should render without crashing', () => {
      const { wrapper } = initCreateEdit()
      expect(wrapper).toBeTruthy()
    })

    it('should render edit form if id is set', () => {
      const { wrapper } = initCreateEdit({ match: { params: { id: '123' } } })
      expect(wrapper).toMatchSnapshot()
    })

    it('should render create form if id is not set', () => {
      const { wrapper } = initCreateEdit({ match: { params: {} } })
      expect(wrapper).toMatchSnapshot()
    })
  })
})
