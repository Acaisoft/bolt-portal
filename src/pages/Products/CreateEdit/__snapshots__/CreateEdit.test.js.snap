// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: Products/CreateEdit rendering should render create form if id is not set 1`] = `
<div>
  Creating a new product
</div>
`;

exports[`page: Products/CreateEdit rendering should render edit form if id is set 1`] = `
<div>
  Editing product with ID=123
</div>
`;
